tensorflow/compiler/tf2tensorrt/convert/ops/tile.cc

```diff
@@ -0,0 +1,169 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#if GOOGLE_CUDA && GOOGLE_TENSORRT
+
+#include "tensorflow/compiler/tf2tensorrt/convert/convert_nodes.h"
+#include "tensorflow/compiler/tf2tensorrt/convert/op_converter_registry.h"
+
+namespace tensorflow {
+namespace tensorrt {
+namespace convert {
+
+class ConvertTile : public OpConverterBase<ConvertTile> {
+ public:
+  explicit ConvertTile(OpConverterParams* params)
+      : OpConverterBase<ConvertTile>(params) {}
+
+  static constexpr std::array<DataType, 3> AllowedDataTypes() {
+    return {DataType::DT_FLOAT, DataType::DT_HALF, DataType::DT_INT32};
+  }
+
+  static constexpr std::array<InputArgSpec, 2> InputSpec() {
+    return std::array<InputArgSpec, 2>{
+        InputArgSpec::Create("input_tensor", TrtInputArg::kBoth),
+        InputArgSpec::Create("weight", TrtInputArg::kBoth)};
+  }
+
+  Status Validate() {
+    const auto &params = *this->params_;
+    const auto& inputs = params.inputs;
+    CHECK_INPUT_SIZE(inputs.size(), 2, params.node_def);
+
+    const auto &repl = inputs.at(1);
+    if (params.use_implicit_batch && repl.is_tensor()) {
+      return errors::InvalidArgument("Conversion for Tile is not implementd for multipliers passed as a tensor");
+    }
+
+    nvinfer1::DataType dtype;
+    const int *multiplies;
+    if (repl.is_weights()) {
+      CHECK_SHAPE_TENSOR(repl.weights().GetTensor());
+      dtype = repl.weights().TrtDType();
+      multiplies = repl.weights().GetPointer<int>();
+    } else {
+      dtype = repl.tensor()->getType();
+      multiplies = nullptr;
+    }
+
+    if (dtype != nvinfer1::DataType::kINT32) {
+      return errors::InvalidArgument("The replication parameter of the ", params.node_def.op(), " operation in ",
+                                     params.node_def.name(), " is expected to be of ",
+                                     DebugString(nvinfer1::DataType::kINT32),
+                                     " type, got ", DebugString(dtype));
+    }
+
+    const auto dims = inputs.at(0).GetTrtDims();
+    const auto nb_dims = dims.nbDims + (params.use_implicit_batch && inputs.at(0).is_tensor()? 1 : 0);
+    if (multiplies) {
```

Comment:
If the multiple is not a weight, we don't need to perform similar check?
We need to make sure that the tensor is one dimension and its number of elements equal to the rank of inputs[0]?
is this related to line 2240-2240 in the test, where we need to fix up the error code?

Reply:
Done, see lines 92-103 of tensorflow/compiler/tf2tensorrt/convert/ops/tile.cc